### PR TITLE
Fix guide workflow, take 2

### PR DESCRIPTION
**Pull request by romac**

### CODE REVIEW

File: .github/workflows/guide.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
       # Only deploy guide when releasing a new version of Hermes
       - name: Deploy to GitHub Pages
         uses: peaceiris/actions-gh-pages@v3
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v0.')
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v0')
```
Comment (mzabaluev) [inline .github/workflows/guide.yml:58]: Still not v1-ready :grin:

Comment (adizere) [reply .github/workflows/guide.yml:58]: hehe

**Review by adizere — APPROVED** [review]
Pre-approved.

File: .github/workflows/guide.yml (hunk 1 above)
Comment (romac) [reply .github/workflows/guide.yml:58]: Hopefully we'll have a better solution for the guide by then, ie. with versioning etc.

File: .github/workflows/guide.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - master
     tags:
-      - v0.*
+      - v[0-9]+.*
```
Comment (mzabaluev) [inline .github/workflows/guide.yml:8]: The dot in the regex matches any symbol, if you need to match literally the dot, you need to escape it?

File: .github/workflows/guide.yml (hunk 1 above)
Comment (mzabaluev) [reply .github/workflows/guide.yml:58]: Maybe if you rely on the workflow trigger being either push to `master` or to a `v[0-9]+\.*` tag, it's enough to discern here that the push was for a tag.

File: .github/workflows/guide.yml (hunk 2 above)
Comment (mzabaluev) [reply .github/workflows/guide.yml:8]:
```suggestion
      - v[0-9]+\.*
```

File: .github/workflows/guide.yml (hunk 1 above)
Comment (romac) [reply .github/workflows/guide.yml]: Good point!

File: .github/workflows/guide.yml (hunk 2 above)
Comment (mzabaluev) [reply .github/workflows/guide.yml:8]: Ah yes, I forgot the regex must match the whole version string, and furthermore my suggestion is broken.

**Review by mzabaluev — APPROVED** [review]

**Pull request merged**